components/motor/gpio/motor_encoder.go

```diff
@@ -316,11 +313,20 @@ func (m *EncodedMotor) directionMovingInLock() float64 {
 func (m *EncodedMotor) SetPower(ctx context.Context, powerPct float64, extra map[string]interface{}) error {
 	m.stateMu.Lock()
 	defer m.stateMu.Unlock()
-	return m.setPower(ctx, powerPct, false)
+	if m.rpmMonitorDone != nil {
+		m.rpmMonitorDone()
+	}
+	if m.loop != nil {
+		m.loop.Stop()
+		m.loop = nil
+	}
+	return m.setPower(ctx, powerPct)
 }
 
 // setPower assumes the state lock is held.
-func (m *EncodedMotor) setPower(ctx context.Context, powerPct float64, internal bool) error {
+func (m *EncodedMotor) setPower(ctx context.Context, powerPct float64) error {
+	// m.stateMu.Lock()
+	// defer m.stateMu.Unlock()
 	dir := sign(powerPct)
 	// If the control config exists, a control loop must exist, so the motor should be allowed to run at a power lower than 10%.
```

Comment:
can we remove this logic? I don't see a reason to arbitrarily assume the motor deadzone is 10%

Reply:
I agree. maybe instead use min powerPct? because with the rover the motors won't really move if given a power below 10%. in that case tho I think we need to have minPowerPct as part of the fragment if it isn't already

Reply:
using a minPowerPct would be fine, but I think we don't need to add it to the fragment. it should be okay to set a power that isn't strong enough for the motors to move.

Reply:
In the setup you described, shouldn't rpmMonitor keep switching between stopped and moving? in which case the user would see 10RPM was too low(or we aren't using a good enough rampRate). I also think its fine to put some minimum power requirement in rpmMonitor.

I just think set power functions should have a minimal amount of logic, because they are supposed to be the closest thing to directly controlling the motors. but I also do not feel strongly about this

Reply:
yeah but that feels like bad behavior in the sense that rpmMonitor should use the minimum of the requested power and the minimum power. but I see your point the user should know to change it in that case but it feels like for the rover a user shouldn't run into this issue

Reply:
I'll try moving the logic outside of setPower, maybe just adding logs of some kind

Reply:
decided to just remove it